src/evaluation/evaluation.service.ts

```diff
@@ -76,27 +76,36 @@ export class EvaluationService {
         await this.returnManager().save(mp);
       }
 
-      for (const id of item.testSumIds) {
-        const ts = await this.returnManager().findOneBy(TestSummary, {
-          testSumIdentifier: id,
-        });
-        ts.evalStatusCode = 'INQ';
-
-        const tsRecord = new Evaluation();
-        tsRecord.evaluationSetIdentifier = set_id;
-        tsRecord.processCode = 'QA';
-
-        if (item.submitMonPlan === false) {
-          tsRecord.statusCode = 'QUEUED';
-        } else {
-          tsRecord.statusCode = 'PENDING';
+      const testSumIdsOrderdList = await this.returnManager()
+        .getRepository(TestSummary)
+        .createQueryBuilder('ts')
+        .select(['ts.test_sum_id'])
+        .where({ testSumIdentifier: In(item.testSumIds) })
+        .orderBy(
+          `(case when ts.test_type_cd='RATA' then 1 when ts.test_type_cd='F2LREF' then 2 when ts.test_type_cd='F2LCHK' then 3 when ts.test_type_cd='FFACC' then 4 when ts.test_type_cd='FFACCTT' then 4 when ts.test_type_cd='PEI' then 4 when ts.test_type_cd='FF2LBAS' then 5 when ts.test_type_cd='FF2LTST' then 6 else null end)`,
+        )
+        .getRawMany();
+
+      if (testSumIdsOrderdList?.length) {
+        for (const testSummary of testSumIdsOrderdList) {
+          const ts = await this.returnManager().findOneBy(TestSummary, {
+            testSumIdentifier: testSummary.test_sum_id,
+          });
+
+          ts.evalStatusCode = 'INQ';
+          const tsRecord = new Evaluation();
+          tsRecord.evaluationSetIdentifier = set_id;
+          tsRecord.processCode = 'QA';
+          if (item.submitMonPlan === false) {
+            tsRecord.statusCode = 'QUEUED';
+          } else {
+            tsRecord.statusCode = 'PENDING';
+          }
+          tsRecord.testSumIdentifier = testSummary.test_sum_id;
```

Comment:
You could use just the first query if you remove the `.select` call here and retrieve all the columns:
```suggestion
      const testSumsOrderedList = await this.returnManager()
        .getRepository(TestSummary)
        .createQueryBuilder('ts')
        .where({ testSumIdentifier: In(item.testSumIds) })
        .orderBy(
          `(case when ts.test_type_cd='RATA' then 1 when ts.test_type_cd='F2LREF' then 2 when ts.test_type_cd='F2LCHK' then 3 when ts.test_type_cd='FFACC' then 4 when ts.test_type_cd='FFACCTT' then 4 when ts.test_type_cd='PEI' then 4 when ts.test_type_cd='FF2LBAS' then 5 when ts.test_type_cd='FF2LTST' then 6 else null end)`,
        )
        .getMany();

      if (testSumsOrderedList?.length) {
        for (const testSummary of testSumsOrderedList) {
          testSummary.evalStatusCode = 'INQ';
          const tsRecord = new Evaluation();
          tsRecord.evaluationSetIdentifier = set_id;
          tsRecord.processCode = 'QA';
          if (item.submitMonPlan === false) {
            tsRecord.statusCode = 'QUEUED';
          } else {
            tsRecord.statusCode = 'PENDING';
          }
          tsRecord.testSumIdentifier = testSummary.testSumIdentifier;
```
Then `testSummary` can be referenced directly.